src/main/scala/beam/sim/BeamHelper.scala

```diff
@@ -344,18 +347,19 @@ trait BeamHelper extends LazyLogging {
   def privateVehicles(
     beamConfig: BeamConfig,
     vehicleTypes: Map[Id[BeamVehicleType], BeamVehicleType]
-  ): TrieMap[Id[BeamVehicle], BeamVehicle] =
+  ): (TrieMap[Id[BeamVehicle], BeamVehicle], mutable.Map[Id[BeamVehicle], Double]) =
```

Comment:
have you considered adding initial state of charge to `BeamVehicle` itself or if it is not possible maybe to create a wrapper for `BeamVehicle` with `Option[Double]` for SoC using either a case class or a tuple + custom type. I think it would be easier to work with than using two separate maps and joining them later.

Reply:
I thought about it. But the initial SoC is mostly related to the scenario and to the vehicle state. Maybe I'm wrong here.